js/client-teambuilder.js

```diff
@@ -2216,25 +2234,16 @@
 			var hpTypes = ['Fighting', 'Flying', 'Poison', 'Ground', 'Rock', 'Bug', 'Ghost', 'Steel', 'Fire', 'Water', 'Grass', 'Electric', 'Psychic', 'Ice', 'Dragon', 'Dark'];
 			var hpType;
 			if (this.curTeam.gen <= 2) {
-				var hpDV = Math.floor(set.ivs.hp / 2);
-				var atkDV = Math.floor(set.ivs.atk / 2);
-				var defDV = Math.floor(set.ivs.def / 2);
-				var speDV = Math.floor(set.ivs.spe / 2);
-				var spcDV = Math.floor(set.ivs.spa / 2);
+				var autoIVs = this.getAutoIVs();
+				var atkDV = Math.floor(autoIVs.atk / 2);
+				var defDV = Math.floor(autoIVs.def / 2);
 				hpType = hpTypes[4 * (atkDV % 4) + (defDV % 4)];
-				var expectedHpDV = (atkDV % 2) * 8 + (defDV % 2) * 4 + (speDV % 2) * 2 + (spcDV % 2);
-				if (expectedHpDV !== hpDV) {
-					set.ivs.hp = expectedHpDV * 2;
-					if (set.ivs.hp === 30) set.ivs.hp = 31;
-					this.$chart.find('input[name=iv-hp]').val(expectedHpDV);
-				}
```

Comment:
I removed the behavior where we automatically adjust the HP here. It will cause problems with the new way we store the IVs, and this behavior is only triggered when there's Hidden Power in the moveset anyway.

Reply:
What's the new plan for auto HP adjustment?

Reply:
I did not add replacement code for this. However, we only trigger this code if the Pokemon knows Hidden Power anyway, so most people never see this code execute in the first place. (If you mean Hidden Power and not hit points, we still adjust the HP type in the moveset after this)